File: Szhchem.Chemistry/Models/MoleculeBase.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Szhchem.Chemistry
{
    public class MoleculeBase
    {
        public string MolFile { get; set; }
        public string IsomericSmiles { get; set; }
        public string CanonicSmiles { get; set; }
        public double ExactMass { get; set; }
        public double Mass { get; set; }
        public string Formula { get; set; }
        public short Charge { get; set; }
        public string InChI { get; set; }
        public string InChIKey { get; set; }
        public MolStat MolStat { get; set; } = new MolStat();
        public long[] Fp4 { get; set; } = new long[16];
        public long[] Fp2 { get; set; } = new long[32];
    }
}

File: molfinder/ClientApp/src/models/MoleculeBase.ts
class MoleculeBase {
    public molFile: string;
    public isomericSmiles: string;
    public canonicSmiles: string;
    public exactMass: number;
    public mass: number;
    public formula: string;
    public charge: number;
    public inChI: string;
    public inChIKey: string;

    constructor(molFile: string, isomericSmiles: string, canonicSmiles: string, exactMass: number, mass: number, formula: string, charge: number, inChI: string, inChIKey: string) {
        this.molFile = molFile;
        this.isomericSmiles = isomericSmiles;
        this.canonicSmiles = canonicSmiles;
        this.exactMass = exactMass;
        this.mass = mass;
        this.formula = formula;
        this.charge = charge;
        this.inChI = inChI;
        this.inChIKey = inChIKey;
    }
}

export default MoleculeBase;